src/aspire/volume/__init__.py

```diff
@@ -317,3 +467,31 @@ def rotated_grids(L, rot_matrices):
 
     # Note we return grids as (Z, Y, X)
     return pts_rot[::-1]
+
+
+def rotated_grids_3d(L, rot_matrices):
+    """
+    Generate rotated Fourier grids in 3D from rotation matrices
+    :param L: The resolution of the desired grids.
+    :param rot_matrices: An array of size k-by-3-by-3 containing K rotation matrices
+    :return: A set of rotated Fourier grids in three dimensions as specified by the rotation matrices.
+        Frequencies are in the range [-pi, pi].
+    """
+
+    grid3d = grid_3d(L, dtype=rot_matrices.dtype)
+    num_pts = L ** 3
+    num_rots = rot_matrices.shape[0]
+    pts = np.pi * np.vstack(
+        [
+            grid3d["x"].flatten(),
+            grid3d["y"].flatten(),
+            grid3d["z"].flatten(),
+        ]
+    )
+    pts_rot = np.zeros((3, num_pts, num_rots), dtype=rot_matrices.dtype)
+    for i in range(num_rots):
+        pts_rot[:, :, i] = rot_matrices[i, :, :] @ pts
+
+    pts_rot = m_reshape(pts_rot, (3, L ** 3 * num_rots))
```

Comment:
I have some suspicion this is just a transpose in this case.  If that is true then we can refactor L491 to the end of the function. The `m_` functions can be a real pain.